src/ArgentPonyWarcraftClient/Models/AssetWithoutFileDataId.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Text.Json.Serialization;
+
+namespace ArgentPonyWarcraftClient
+{
+    /// <summary>
+    /// A media asset.
+    /// </summary>
+    public record AssetWithoutFileDataId
```

Comment:
> I'm thinking maybe we should make the `FileDataId` property on the original `Asset` class a nullable, i.e. `long? FileDataId { get; init; }`. Then it reduces the maintenance effort if an asset has more fields added that are shared between both models. What are your thoughts?

Yes, I'm of divided mind about that.  It would reduce the number of types and avoid awkward type names with `Without` in the name.  On the other hand, we would sometimes be returning objects with properties that can (for their usage in that particular method) never have a non-null value.  I'm not really happy with either scenario.  If you'd like to tip the scales one way or another, go for it.